### PR TITLE
Transitory Edge Analysis

### PR DESCRIPTION
**Pull request by mjohnson541**
RMS companion to https://github.com/ReactionMechanismGenerator/RMG-Py/pull/2317. Implements transitory edge analysis.

### CODE REVIEW

File: src/EdgeAnalysis.jl [hunk 1]
```diff
@@ -606,7 +606,45 @@ function identifyobjects!(sim,corespcsinds,corerxninds,edgespcsinds,
...
         tempnewobjecttype = []
     end
-    
+
+
+    if length(transitorydict) > 0 && !firsttime
+        transitoryoutdict = Dict()
+        TS = transitorysensitivitiesfulltrapezoidal(sim,t;normalized=false)
+        TS .*= sim.p'
+        TS = TS[:,length(sim.names)+1:end]
```
Comment (hwpang) [inline src/EdgeAnalysis.jl:657]: Could you add @views here

File: src/EdgeAnalysis.jl [hunk 2]
```diff
@@ -42,31 +42,50 @@ end
...
+"""
+Generate a state vector appropriate for both the edge and core from the core state vector
+"""
+function getycoreedge(y,coretoedgespcmap,edgelen,react,coreedgedomain)
+    ycoreedge = zeros(edgelen)
+    for (coreind,edgeind) in coretoedgespcmap
+        ycoreedge[edgeind] = y[coreind]
+    end
+    for j in 3:length(coreedgedomain.indexes)
+        ycoreedge[coreedgedomain.indexes[j]] = y[react.domain.indexes[j]]
```
Comment (hwpang) [inline src/EdgeAnalysis.jl:87]: Could you add @inbounds to several places in this PR that uses indexing?

File: src/EdgeAnalysis.jl [hunk 3]
```diff
@@ -657,8 +657,9 @@ function identifyobjects!(sim,corespcsinds,corerxninds,edgespcsinds,
         TS = TS[:,length(sim.names)+1:end]
         for (spcname,tol) in transitorydict
             spcind = findfirst(isequal(spcname),sim.names)
+            tsscale = maximum(abs.(TS[spcind,:]))
```
Comment (hwpang) [inline src/EdgeAnalysis.jl:660]: Can you add @views here

File: src/EdgeAnalysis.jl (hunk 1 above)
Comment (mjohnson541) [reply src/EdgeAnalysis.jl:657]: I don't think @views will work here in the only line with a slice we're defining a new matrix so it has to create a new matrix. I have added an @views to the tsscale line.

File: src/EdgeAnalysis.jl (hunk 3 above)
Comment (mjohnson541) [reply src/EdgeAnalysis.jl:660]: Done

File: src/EdgeAnalysis.jl (hunk 2 above)
Comment (mjohnson541) [reply src/EdgeAnalysis.jl:87]: Done

**Comment by mjohnson541** [comment]
Okay I think I've fixed all of the issues and I've fixed the CI workflow.

**Pull request merged**